### PR TITLE
make label name validation optional

### PR DESCRIPTION
**Pull request by replay**
introduces a new configuration parameter to allow the user to disable label name validation in the distributor.
the option to disable label name validation is useful for users who want to store non-prometheus data in cortex.

Fixes https://github.com/cortexproject/cortex/issues/2770

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pstibrany** [comment]
Would it make sense to make internal only, and not visible for the user? No Cortex user should really use this.

**Comment by replay** [comment]
> Would it make sense to make internal only, and not visible for the user? No Cortex user should really use this.

Yeah i guess you're right. If a cortex user does need it we can still make it visible for the user in the future.

**Comment by replay** [comment]
I wasn't sure where the best place is to put this flag if it isn't supposed to be visible to the user. 
I've now made it a property of the Distributor config, do you think it would be better to keep the setting in `validation.Overrides` but just not let the user configure this flag ?

File: pkg/distributor/distributor.go [hunk 1]
```diff
@@ -154,6 +154,9 @@ type Config struct {
 
 	// for testing
 	ingesterClientFactory ring_client.PoolFactory `yaml:"-"`
+
+	// when true the distributor does not validate labels at ingest time
```
Comment (pracucci) [inline pkg/distributor/distributor.go:158]: Would you mind adding something like "Cortex doesn't directly use it (and should never use it) but this feature is used by other projects built on top of it.".

Comment (replay) [reply pkg/distributor/distributor.go:158]: Sure, adding that

**Comment by replay** [comment]
> Yes, I assume you will need to have similar validation in place that makes sense for your use-case.

Indeed, we'll have validation based on our own rules at ingest time

**Comment by replay** [comment]
Thx for the approvals. It appears I can't merge the PR myself because I do not have write access to the repo. Could one of you do it please? Thank you.

**Pull request merged**